src/NuGet.Services.AzureSearch/SearchService/Models/ParameterUtilities.cs

```
@@ -45,5 +55,31 @@ public static bool ParseIncludeSemVer2(string semVerLevel)
                 return semVerLevelVersion >= SemVer2Level;
             }
         }
+
+        public static IReadOnlyList<string> ParseFrameworks(string frameworks)
+        {
+            return frameworks == null
+                                    ? (IReadOnlyList<string>)Array.Empty<string>()
+                                    : frameworks
+                                        .Split(',')
+                                        .Select(f => f.ToLowerInvariant().Trim())
+                                        .Where(f => f != String.Empty)
+                                        .Where(f => FrameworkGenerationIdentifiers.Contains(f))
+                                        .ToList();
+        }
+
+        public static IReadOnlyList<string> ParseTfms(string tfms)
+        {
+            return tfms == null
+                            ? (IReadOnlyList<string>)Array.Empty<string>()
+                            : tfms
+                                .Split(',')
+                                .Select(f => f.Trim())
+                                .Where(f => f != String.Empty)
+                                .Select(f => NuGetFramework.Parse(f))
```

Take a look at the code of `NuGetFramework`. There are some cases that throw instead of returning unspecified.
https://github.com/NuGet/NuGet.Client/blob/d1620d3bc77030d05e7e76b5adf06c33c4dc8374/src/NuGet.Core/NuGet.Frameworks/NuGetFrameworkFactory.cs#L137-L144

https://github.com/NuGet/NuGet.Client/blob/d1620d3bc77030d05e7e76b5adf06c33c4dc8374/src/NuGet.Core/NuGet.Frameworks/NuGetFrameworkFactory.cs#L247-L250

We may need to catch `ArgumentException` 😨. UTs should probably cover these to verify behavior over time.

---

I've done this and added tests as well. Let me know how it looks